apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c

```diff
@@ -596,6 +596,31 @@ static void utc_audio_pcm_format_to_bits_n(void)
 	TC_SUCCESS_RESULT();
 }
 
+/**
+* @testcase         audio_pcm_drop_p
+* @brief            drop all the buffers which are being processed and stop the device
+* @scenario         call pcm_drop after readi and check that audio after pcm_drop print statement is recorded.
+* @apicovered       pcm_drop
+* @precondition     NA
+* @postcondition    utc_audio_pcm_readi_p should be executed immediately after this function
+*/
+
+static void utc_audio_pcm_drop_p(void)
+{
+	int ret;
+	char *buffer = malloc(pcm_frames_to_bytes(g_pcm, pcm_get_buffer_size(g_pcm)));
+	TC_ASSERT_NEQ_CLEANUP("pcm_readi", buffer, NULL, clean_all_data(0, NULL));
+	printf("Staring pcm device to capture\n");
+	ret = pcm_readi(g_pcm, buffer, pcm_get_buffer_size(g_pcm));
+	TC_ASSERT_GEQ("pcm_drop", ret, 0);
```

Comment:
not cleanup for every TC_ASSERT?

Reply:
No, drop_p and readi_p tc's are closely linked to each other, so the cleanup function cannot be called here as pcm object is need by readi_p tc to have a successful result.

Reply:
Then why do you call cleanup at above TC_ASSERT macro? and even if pcm is used at next tc, you should free allocated memory, buffer. Isn't it?

Reply:
I missed freeing the buffer in error case, I will add that.I did cleanup in the first case because the error is caused by malloc(not a part we want to test) which may occur with high probability in the coming tc's.So I felt it is better to safely close the audio device and let the coming tcs fail too.